isamples_metadata/GEOMETransformer.py

```diff
@@ -0,0 +1,362 @@
+
+    def sample_identifier_value(self) -> typing.AnyStr:
+        return "/"
+
+    def sample_description(self) -> typing.AnyStr:
+        # TODO: implement
+        # [concatenate key:value pairs from other content in record/ that is not mapped, e.g.
+        # 'voucherCatalogNumber:USNM:Fish:433156'; record/preservative, ]
+        return ""
+
+    def has_context_categories(self) -> typing.List[typing.AnyStr]:
+        # TODO: implement
+        # ["[infer from locality and taxon names]"]
+        return []
+
+    def has_material_categories(self) -> typing.List[typing.AnyStr]:
+        # TODO: implement
+        # ["'Organic material' unless record/entity, record/basisOfRecord, or record/collectionCode indicate otherwise"]
+        return ["Organic material"]
+
+    def has_specimen_categories(self) -> typing.List[typing.AnyStr]:
+        # TODO: implement
+        # ["'Whole organism'  unless record/entity, record/basisOfRecord, or record/collectionCode indicate otherwise"]
+        return ["Whole organism"]
+
+    def informal_classification(self) -> typing.AnyStr:
+        main_record = self._source_record_main_record()
+        informal_classification = main_record.get("scientificName")
+        if informal_classification is None:
+            pieces = []
+            genus = main_record.get("genus")
+            if genus is not None:
+                pieces.append(genus)
+            epithet = main_record.get("specificEpithet")
+            if epithet is not None:
+                pieces.append(epithet)
+            informal_classification = " ".join(pieces)
+        return informal_classification
+
+    def _place_names(self, only_general: bool):
+        parent_record = self._source_record_parent_record()
+        if parent_record is not None:
+            place_names = []
+            if not only_general:
+                if "county" in parent_record:
+                    place_names.append(parent_record["county"])
+                if "locality" in parent_record:
+                    place_names.append(parent_record["locality"])
+            if "stateProvince" in parent_record:
+                place_names.append(parent_record["stateProvince"])
+            if "island" in parent_record:
+                place_names.append(parent_record["island"])
+            if "islandGroup" in parent_record:
+                place_names.append(parent_record["islandGroup"])
+            if "country" in parent_record:
+                place_names.append(parent_record["country"])
+            if "continentOcean" in parent_record:
+                place_names.append(parent_record["continentOcean"])
+            return place_names
+        return []
+
+    def keywords(self) -> typing.List[typing.AnyStr]:
+        # "JSON array of values from record/ -order, -phylum, -family, -class, and parent/ -country, -county,
+        # -stateProvince, -continentOcean... (place names more general that the locality or most specific
+        # rank place name) "
```

Comment:
This yielded slightly different data from the test examples, but I tried to follow the explanation rather than the examples where the paths diverged.